### PR TITLE
Warnings support

### LINKED ISSUE #92: xdist does not transfer warnings to the master
pytest-dev/pytest#1996 and pytest-dev/pytest#602

example from the pytest issue:

Warnings raised with `request.node.warn` does not show warnings. `-rw` does not give any warning output either.

Here is the same test run without and with xdist:

```
$ py.test unit_tests/
......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
2166 passed, 78 warnings in 46.73 seconds
```

```
$ py.test unit_tests/ -n1
gw0 [2166]
scheduling tests via LoadScheduling
......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
2166 passed in 48.02 seconds
```

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #92

### CODE REVIEW

**Comment by nicoddemus** [comment]
Unfortunately the test must be skipped on pytest versions older than 3.1. Bummer that we don't run 3.1 or older on CI yet due to #169.

File: testing/acceptance_test.py [hunk 1]
```diff
@@ -339,6 +339,19 @@ def test_func():
             "E       assert 0",
         ])
 
+    @pytest.mark.parametrize('n', ['-n0', '-n1'])
+    @pytest.mark.skipif(pytest.__version__ < "3.1", reason='pytest warnings requires >= 3.1')
```
Comment (RonnyPfannschmidt) [inline testing/acceptance_test.py]: please use version objects for version comparisation

**Comment by RonnyPfannschmidt** [comment]
i#ll begin by doin the travis fixup

**Comment by RonnyPfannschmidt** [comment]
will need a rebase after merging #210

**Comment by nicoddemus** [comment]
Rebased and used `pkg_resources` to parse pytest's version as suggested.

**Comment by RonnyPfannschmidt** [comment]
we should document that we cant tell what slave a warning is from

**Pull request merged**